sql/src/main/java/io/crate/execution/engine/collect/files/FileReadingIterator.java

```diff
@@ -84,6 +87,7 @@
     private long currentLineNumber;
     private LineContext lineContext;
     private final Row row;
+    private Optional<String> csvHeader;
```

Comment:
We have recently tried to remove `Optional` and use @Nullable annotations for optional values. The Optional API has been designed for function return values only, it shouldn't be used for class fields.

Reply:
I would strongly suggest to remove this and just have the InputFormat here. The csv header should be part of the csv LineContext.

Reply:
Noted. Curious about why there's been an effort to remove the use of `Optional`?